src/main/java/org/wso2/carbon/connector/FileFtpOverProxy.java

```diff
@@ -0,0 +1,236 @@
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.wso2.carbon.connector;
+
+import org.apache.axiom.om.OMElement;
+import org.apache.commons.lang.StringUtils;
+import org.apache.commons.logging.Log;
+import org.apache.commons.logging.LogFactory;
+import org.apache.commons.net.ftp.FTP;
+import org.apache.commons.net.ftp.FTPClient;
+import org.apache.commons.net.ftp.FTPConnectionClosedException;
+import org.apache.commons.net.ftp.FTPHTTPClient;
+import org.apache.commons.net.ftp.FTPReply;
+import org.apache.synapse.MessageContext;
+import org.codehaus.jettison.json.JSONException;
+import org.wso2.carbon.connector.core.AbstractConnector;
+import org.wso2.carbon.connector.core.Connector;
+import org.wso2.carbon.connector.core.util.ConnectorUtils;
+import org.wso2.carbon.connector.util.FileConstants;
+import org.wso2.carbon.connector.util.ResultPayloadCreate;
+
+import javax.xml.stream.XMLStreamException;
+import java.io.ByteArrayInputStream;
+import java.io.IOException;
+import java.io.InputStream;
+
+/**
+ * FTP Over Proxy
+ */
+public class FileFtpOverProxy extends AbstractConnector implements Connector {
+    private static final Log log = LogFactory.getLog(FileFtpOverProxy.class);
+
+    public void connect(MessageContext messageContext) {
+        String proxyHost = (String) ConnectorUtils
+                .lookupTemplateParamater(messageContext, FileConstants.PROXY_HOST);
+        String proxyPort = (String) ConnectorUtils
+                .lookupTemplateParamater(messageContext, FileConstants.PROXY_PORT);
+        String proxyUsername = (String) ConnectorUtils
+                .lookupTemplateParamater(messageContext, FileConstants.PROXY_USERNAME);
+        String proxyPassword = (String) ConnectorUtils
+                .lookupTemplateParamater(messageContext, FileConstants.PROXY_PASSWORD);
+        String ftpServer = (String) ConnectorUtils
+                .lookupTemplateParamater(messageContext, FileConstants.FTP_SERVER);
+        String ftpPort = (String) ConnectorUtils
+                .lookupTemplateParamater(messageContext, FileConstants.FTP_PORT);
+        String ftpUsername = (String) ConnectorUtils
+                .lookupTemplateParamater(messageContext, FileConstants.FTP_USERNAME);
+        String ftpPassword = (String) ConnectorUtils
+                .lookupTemplateParamater(messageContext, FileConstants.FTP_PASSWORD);
+
+        boolean resultStatus =
+                ftpOverHttp(proxyHost, proxyPort, proxyUsername, proxyPassword, ftpServer, ftpPort,
+                        ftpUsername, ftpPassword, messageContext);
+        generateResult(messageContext, resultStatus);
+    }
+
+    /**
+     * Generate the result.
+     *
+     * @param messageContext The message context that is generated for processing the file
+     * @param resultStatus   true/false
```

Comment:
Can you elaborate this more ? true/false in a boolean would be understandable from it's type

Reply:
Fixed in wso2-extensions/esb-connector-file#28